### ISSUE TITLE
Notification opening favorites or direct search (option)

### ISSUE THREAD

**Issue by falzonv** [falzonv/discreet-launcher#75]
This ticket is opened following suggestion from @xi7 in ticket #67:

> and notification... favorites or direct search

---

I will have to check, if this can be done without too much duplicate code I am ok to implement this option.

**Comment by xi7** [comment]
some game phone have triggers , its possible select it and open direct search or notification or any other physical button ?

**Comment by falzonv** [comment]
Hello,

No sorry, this is not something I plan to implement.

I also realize that the direct search in the notification will not be possible because I can only launch an "Activity" from the notification (this was one of the reasons for the try to convert Search from a "PopupWindow" to an "Activity" in v4.4.0, but we saw that it was not working correctly as an "Activity").

Best regards.

**Comment by xi7** [comment]
![Screenshot_2021-07-19-17-17-28](https://user-images.githubusercontent.com/67331785/126221772-c84ab27b-8ee0-47e2-afc3-ad204abbf9ae.png)

https://apkpure.com/br/talking-translator/com.translate.talkingtranslator

notification organized you can check improvement future.

**Comment by xi7** [comment]
you can reopen and update the status bar, copy this method working very well **without space hide icon.**

also u can use direct search bar, direct flashlight (the color change very good) , direct notification folder....

**Comment by falzonv** [comment]
The notifications under the line only exist in older Android versions.
With more recent versions (starting with Android 7 if I remember well), these notifications are displayed very small and need an extra click before accessing them (even when the notification has only one action like now).

It is the same with buttons in notifications: on some Android versions it looks ok, but with others Android versions it looks awful and sometimes is not even usable (ex: buttons going out of the notification layout).

=> Since Discreet Launcher supports Android 5 to 11, I want to keep its code simple and avoid version-specific tricks as much as possible (that is, only use them to solve a version-specific bug like in #60).

> also u can use direct search bar, direct flashlight (the color change very good) , direct favorites folder....

As I explained in my previous message, the direct search will not be possible due to technical reasons ("Activity" required but search works correctly only with "PopupWindow" ; I made other tests after #86 but this is always slower with an "Activity").

Implementing a flashlight is out of scope and will not be done.

For me it is better to have simply an access to favorites. Then the user puts whatever (s)he wants in the favorites: it could be the Search to have a virtual "All apps" folder, a flashlight app like [Torchie](https://f-droid.org/en/packages/in.blogspot.anselmbros.torchie/), any folder, etc.
